CoinDock-api/app/Http/Controllers/V1/UserController.php

```diff
@@ -13,32 +15,53 @@ class UserController extends AccessTokenController
 {
   use BuildPassportTokens;
 
-  /**
-   * Show the form for creating a new resource.
-   *
-   * @return \Illuminate\Http\Response
-   */
-  public function create(CreateUserRequest $request)
-  {
-    $user = new User();
-    $user->store($request);
-
-    $response = $this->requestPasswordGrant($request);
-
-    return response(
-      [
-        'status' => 'success', 'message' => 'Success! User registered.',
-        'token' => $response['access_token']
-
-      ],
-      Response::HTTP_OK,
-      [
-        'Access-Token' => $response['access_token'],
-        'Refresh-Token' => $response['refresh_token'],
-        'Expires-In' => $response['expires_in'],
-      ],
-    );
-  }
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create(CreateUserRequest $request)
+    {
+        $user = new User();
+        $userData = $user->store($request);
+
+        $response = $this->requestPasswordGrant($request);
+        
+        return response(
+            ['status' => 'success', 'message' => 'Success! User registered.'
+                
+        ],
+            Response::HTTP_OK,
+            [
+                'Access-Token' => $response['access_token'],
+                'Refresh-Token' => $response['refresh_token'],
+                'Expires-In' => $response['expires_in'],
+            ],
+        );
+    }
+
+    public function changePassword(updatePasswordRequest $request, User $user)
+    {
+        return $user->changePassword($request,$user);
+    }
+
+
+    public function updateProfile(updateProfileRequest $request, User $user)
+    {
+        return $user->updateProfile($request,$user);
+
+    }
+
+    //list of user titles 
+    public function usersTitles(){
+      $titles = ['Mr.','Ms.','Mrs.','Mx.'];
```

Comment:
drive title values from model, please.

It's related to columns value so added a static function like:

 public static function getSupportedTitles(): array
    {
        return ['Mr', 'Mrs', 'Miss', 'Ms', 'Dr'];
    } 
in user model

call above function in controller like User::getSupportedTitles()